src/IO/S3/URI.cpp

```diff
@@ -18,6 +18,26 @@
 namespace DB
 {
 
+struct UriConverter
+{
+    static void modifyURI(Poco::URI & uri)
+    {
+        std::string domain;
+        if (uri.getScheme() == "s3")
+            domain = S3_DOMAIN;
+        else if (uri.getScheme() == "oss")
+            domain = OSS_DOMAIN;
+        else if (uri.getScheme() == "gs")
+            domain = GCS_DOMAIN;
+        uri.setScheme("https");
+        uri.setHost(uri.getHost() + domain);
+    }
+private:
+    static constexpr auto S3_DOMAIN = ".s3.amazonaws.com";
```

Comment:
When a constant is used once, it's ok to have it inline, without a separate constant definition.